streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/AppMaster.scala

```diff
@@ -42,30 +42,36 @@ class AppMaster(appContext : AppMasterContext, app : Application)  extends Appli
   private val LOG: Logger = LogUtil.getLogger(getClass, app = appId)
   LOG.info(s"AppMaster[$appId] is launched by $username $app xxxxxxxxxxxxxxxxx")
 
-  private val (clockService, taskManager, executorManager) = {
+  private val (taskManager, executorManager) = {
     val dag = DAG(userConfig.getValue[Graph[TaskDescription, Partitioner]](AppDescription.DAG).get)
-    val clockService : ActorRef = {
-      val store = new InMemoryAppStoreOnMaster(appId, masterProxy)
-      context.actorOf(Props(new ClockService(dag, store)))
-    }
+
     val executorManager = context.actorOf(ExecutorManager.props(userConfig, appContext))
-    val taskManager = context.actorOf(Props(new TaskManager(appContext, dag, clockService, executorManager, self)))
-    (clockService, taskManager, executorManager)
+
+    val store = new InMemoryAppStoreOnMaster(appId, appContext.masterProxy)
+    val clockService = context.actorOf(Props(new ClockService(dag, store)))
+
+    val taskScheduler: TaskScheduler = new TaskSchedulerImpl(appId, context.system.settings.config)
+    val taskManager = context.actorOf(Props(new TaskManager(appContext.appId, dag,
+      taskScheduler, executorManager, clockService, self)))
+    (taskManager, executorManager)
   }
```

Comment:
Here ExecutorManager has the reference of TaskManager and TaskManager has the reference of ExecutorManager, I'm wondering is it better to let AppMaster be the mediator of messages across them so that if we want to extract a general AppMaster, these managers could be independent to be reused.